api/app/controllers/spree/api/v2/storefront/taxons_controller.rb

```diff
@@ -0,0 +1,44 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class TaxonsController < ::Spree::Api::V2::BaseController
+          def index
+            render json: serialize_taxons_collection(collection), status: 200
+          end
+
+          def show
+            render json: serialize_taxon_resource(resource), status: 200
+          end
+
+          private
+
+          def dependencies
+            {
+              taxons_finder:                Spree::Taxons::Find,
+              taxon_resource_serializer:    Spree::V2::Storefront::TaxonSerializer,
+              taxons_collection_serializer: Spree::V2::Storefront::TaxonSerializer
+            }
+          end
+
+          def collection
+            base_scope = Spree::Taxon.includes(:parent, :children).accessible_by(current_ability, :read)
+            dependencies[:taxons_finder].new(base_scope, params).call
+          end
+
+          def resource
+            Spree::Taxon.find_by(id: params[:id]) || Spree::Taxon.find_by(permalink: params[:id])
```

Comment:
I think we should use `find_by || find` so that the error is raised whenever there's no record.
Currently it'll just return `nil`.